drivers/uktty/com/Config.uk

```diff
@@ -0,0 +1,131 @@
+menuconfig LIBUKTTY_COM
+	bool "COM serial port drivers"
+	depends on ARCH_X86_64
+	select LIBUKCONSOLE
+
+if LIBUKTTY_COM
+menuconfig LIBUKTTY_COM1
+	bool "Enable COM1 driver (0x3f8)"
+	default y
+
+if LIBUKTTY_COM1
+choice
+	prompt "COM1 port baud rate"
+	default LIBUKTTY_COM1_BAUD_115200
+
+config LIBUKTTY_COM1_BAUD_115200
+	bool "115200"
+
+config LIBUKTTY_COM1_BAUD_57600
+	bool "57600"
+
+config LIBUKTTY_COM1_BAUD_38400
+	bool "38400"
+
+config LIBUKTTY_COM1_BAUD_19200
+	bool "19200"
+endchoice
+endif
+
+menuconfig LIBUKTTY_COM2
+	bool "Enable COM2 driver (0x2f8)"
+
+if LIBUKTTY_COM2
+choice
+	prompt "COM2 port baud rate"
+	default LIBUKTTY_COM2_BAUD_115200
+
+config LIBUKTTY_COM2_BAUD_115200
+	bool "115200"
+
+config LIBUKTTY_COM2_BAUD_57600
+	bool "57600"
+
+config LIBUKTTY_COM2_BAUD_38400
+	bool "38400"
+
+config LIBUKTTY_COM2_BAUD_19200
+	bool "19200"
+endchoice
+endif
+
+menuconfig LIBUKTTY_COM3
+	bool "Enable COM3 driver (0x3e8)"
+
+if LIBUKTTY_COM3
+choice
+	prompt "COM3 port baud rate"
+	default LIBUKTTY_COM3_BAUD_115200
+
+config LIBUKTTY_COM3_BAUD_115200
+	bool "115200"
+
+config LIBUKTTY_COM3_BAUD_57600
+	bool "57600"
+
+config LIBUKTTY_COM3_BAUD_38400
+	bool "38400"
+
+config LIBUKTTY_COM3_BAUD_19200
+	bool "19200"
+endchoice
+endif
+
+menuconfig LIBUKTTY_COM4
+	bool "Enable COM4 driver (0x2e8)"
+
+if LIBUKTTY_COM4
+choice
+	prompt "COM4 port baud rate"
+	default LIBUKTTY_COM4_BAUD_115200
+
+config LIBUKTTY_COM4_BAUD_115200
+	bool "115200"
+
+config LIBUKTTY_COM4_BAUD_57600
+	bool "57600"
+
+config LIBUKTTY_COM4_BAUD_38400
+	bool "38400"
+
+config LIBUKTTY_COM4_BAUD_19200
+	bool "19200"
+endchoice
+endif
+
+choice
+    prompt "Early COM device"
+    default LIBUKTTY_COM1_EARLY
+    help
+        The early COM device will be initialized in the early
+        init table and it will have the STDIN and STDOUT flags
+        set. Other devices can still be enabled, but they won't
+        will be initialized later in the init table.
+
+config LIBUKTTY_COM1_EARLY
+    bool "COM1"
+    imply LIBUKTTY_COM1
```

Comment:
I think it is better to do a `depends on` instead of `imply`. Problem right now: You can configure a COM port as early console and then unselect the COM port driver which I would consider broken configuration state. I first thought about `select` but `depends on` would allow us to select the first available COM device as early console as default. As soon the driver gets unselected, it should automatically jump back to `None`. With this, a platform could just enable early console initialization and the enabled COM port (by the platform) with the lowest number is selected automatically on a fresh config. Unfortunately, a platform cannot set a choice option. Only boolean options can be set with KConfig.

I am thinking of this (untested):
```
menuconfig LIBUKTTY_COM1
	bool "Enable COM1 driver (0x3f8)"
# <...>

menuconfig LIBUKTTY_COM2
	bool "Enable COM2 driver (0x2f8)"
# <...>

config LIBUKTTY_COM_EARLY_CONSOLE
    bool "Initialize early console"
    depends on LIBUKTTY_COM1_EARLY || LIBUKTTY_COM2_EARLY || ...

choice
    prompt "Console device"
    default LIBUKTTY_COM4_EARLY if !LIBUKTTY_COM1 && !LIBUKTTY_COM2 && !LIBUKTTY_COM3 && LIBUKTTY_COM4
    default LIBUKTTY_COM3_EARLY if !LIBUKTTY_COM1 && !LIBUKTTY_COM2 && LIBUKTTY_COM3
    default LIBUKTTY_COM2_EARLY if !LIBUKTTY_COM1 && LIBUKTTY_COM2
    default LIBUKTTY_COM1_EARLY
    depends on LIBUKTTY_COM_EARLY_CONSOLE # This should deprecate the "None" option

config LIBUKTTY_COM1_EARLY
    bool "COM1"
    depends on LIBUKTTY_COM1
    
config LIBUKTTY_COM2_EARLY
    bool "COM2"
    depends on LIBUKTTY_COM2

config LIBUKTTY_COM3_EARLY
    bool "COM3"
    depends on LIBUKTTY_COM3

config LIBUKTTY_COM4_EARLY
    bool "COM4"
    depends on LIBUKTTY_COM4
```

Reply:
I saw that the driver still compiles if all console devices are unselected: very good because we want to avoid that any possible configuration causes a broken build. I think it is good this way and we do not need to automatically confiugre a default device because a platform's `Config.uk` would do that.